src/hexkit/providers/mongodb/provider.py

```diff
@@ -421,6 +402,36 @@ class MongoDbConfig(BaseSettings):
     )
 
 
+ClientType = TypeVar("ClientType", AsyncIOMotorClient, MongoClient)
+
+
+def get_configured_mongo_client(
```

Comment:
Should we provide this as a resource manager class `ConfiguredMongoClient` that auto-closes the client, but with an extra get() method that allows getting the client without auto-close? Could be useful in some cases (e.g. for one of the tests).